wireapp/wire-android-reloaded
# feat: generate build configs [AR-1463]

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-1463" title="AR-1463" target="_blank"><img alt="Task" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10818?size=medium" />AR-1463</a>  Build config flags
  </td></table>
  <br />
 






















----
#### PR Submission Checklist for internal contributors

- The **PR Title**

- The **PR Description**
----

# What's new in this PR?

### Issues

To make kalium aware of the build configs we needed to generate and update the core logic with the generated values


### Solutions

generated build configs from the default.json and add implementation for a custom remote repositories for a custom config builds 
then pass those generated values to kalium

### Dependencies

Needs releases with:

- https://github.com/wireapp/kalium/pull/557

### Testing

#### How to Test

build the app and check the BuildConfig if it was updated with the values from default.json
check the getBuildConfigUseCase if it's returning the generated values 

----
#### PR Post Merge Checklist for internal contributors

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.
